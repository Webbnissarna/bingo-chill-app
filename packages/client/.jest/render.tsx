import React from "react";
import { render as rtlRender } from "@testing-library/react";

export function render(ui: React.ReactElement) {
  return rtlRender(<>{ui}</>);
}
