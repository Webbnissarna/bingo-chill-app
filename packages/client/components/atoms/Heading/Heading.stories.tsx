import type { Story, StoryDefault } from "@ladle/react";
import Heading from ".";

interface StoryProps {
  text: string;
}

function StoryComponent({ text }: StoryProps) {
  return <Heading>{text}</Heading>;
}

export const Default: Story<StoryProps> = StoryComponent;

Default.args = {
  text: "Hello World",
};

Default.argTypes = {
  text: {
    control: { type: "text" },
  },
};

export default {
  title: "Atoms / Heading",
} satisfies StoryDefault;
