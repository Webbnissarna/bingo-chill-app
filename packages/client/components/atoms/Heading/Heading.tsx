interface HeadingProps {
  children: string;
}

export default function Heading({ children }: HeadingProps): JSX.Element {
  return <h1 className="font-heading text-snowStorm-2 text-5xl">{children}</h1>;
}
