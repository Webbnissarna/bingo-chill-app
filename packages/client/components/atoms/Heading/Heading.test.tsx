import { render } from "../../../.jest/render";
import Heading from ".";

describe("Heading", () => {
  it("renders", () => {
    const { asFragment } = render(<Heading>Hello World</Heading>);

    expect(asFragment()).toMatchSnapshot();
  });
});
