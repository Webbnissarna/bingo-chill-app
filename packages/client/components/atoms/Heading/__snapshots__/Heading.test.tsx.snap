// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading renders 1`] = `
<DocumentFragment>
  <h1
    class="font-heading text-snowStorm-2 text-5xl"
  >
    Hello World
  </h1>
</DocumentFragment>
`;
